

simulatorRate: 50 # Hz

useMap: True
useLidar: True

map:
  path: ""
  origin: [0, 0, 0] # m
  # path: "maps/RAFall23GP.png"
  # origin: [-21.000000, -3.850000, 0.000000] # m
  publishMap: True
  mapTopic: "/map"
  mapFrame: "map"
  mapRate: 1 # Hz
  resolution: 0.025000 # m/cell
  occupied_thresh: 0.196
  wallBuffer: 0.15 # m
  obstacles: 
    timeout: 5 # s
    list: [] # list of obstacles
    # list: # list of obstacles
    #   - [[0.74, 1.53],[1.1, 0.67],[2.06, 0.78],[1.7, 1.97]]
    #   - [[7.0, 0.85],[7.75, 1.04],[7.02, 2.18],[6.42, 1.68]]
    #   - [[4.73, 5.65],[4.92, 3.87],[5.77, 4.22],[5.94, 5.51]]
    #   - [[6.36, 10.34],[6.79, 9.27],[7.86, 9.11],[7.44, 10.86]]
    #   - [[-1.17, 12.62],[-1.38, 11.46],[-0.36, 10.96],[-0.26, 12.31]]
    #   - [[-8.79, 14.41],[-9.37, 13.27],[-8.6, 12.87],[-7.73, 13.73]]
    #   - [[-13.53, 11.35],[-13.22, 10.48],[-12.12, 10.71],[-12.39, 11.94]]
    #   - [[-7.77, 8.98],[-7.35, 7.88],[-6.1, 7.98],[-6.48, 8.88]]
    #   - [[-5.42, 3.16],[-5.08, 1.85],[-4.13, 2.76],[-5.23, 3.39]]
    #   - [[-8.87, 4.57],[-8.31, 3.97],[-7.21, 4.57],[-8.08, 5.57]]
    #   - [[-15.9, 1.7],[-14.97, 0.79],[-14.34, 1.47],[-14.92, 2.37]]
    #   - [[-15.9, 8.32],[-15.09, 7.78],[-14.49, 8.48],[-15.47, 9.11]]
    #   - [[-19.77, 2.6],[-18.77, 0.97],[-18.21, 2.01],[-18.75, 3.07]]
    #   - [[-8.83, -0.44],[-8.89, -1.61],[-7.21, -1.75],[-7.35, -0.5]]
    #   - [[-4.42, -1.52],[-4.04, -2.29],[-2.92, -2.17],[-3.03, -1.21]]
  checkpoints: 
    threshold: 1 # m
    list: [] # list of obstacles
    # list:  # list of checkpoints
    #   - [-0.28, 0.1]
    #   - [5.9, 4.91]
    #   - [-0.63, 12.46]
    #   - [-12.39, 13.0]
    #   - [-4.42, 5.55]
    #   - [-8.62, 1.26]
    #   - [-14.57, 2.03]
    #   - [-17.42, 10.4]
    #   - [-16.15, 0.08]
    #   - [-6.79, -1.84]

cars: ["car_1"]
  
car_1:
  amcl: False
  invertSteering: True
  WB: 0.324 # wheel base (m)
  steeringMax: 0.6 # max steering angle (rad)
  lidar:
    updateRate: 40 # Hz
    angleMin: -2.35619 # -135 degrees
    angleMax: 2.35619 # 135 degrees
    angleIncrement: 0.00436332 # 0.25 degrees
    rangeMin: 0.1 # m
    rangeMax: 30.0 # m
  imu:
    linear_acceleration_variance: [0.01, 0.01, 0.01] # m^2/s^4
    angular_velocity_variance: [0.01, 0.01, 0.01] # rad^2/s^2
    orientation_variance: [0.001, 0.001, 0.001] # rad^2
